enginetest/queries/index_queries.go

```diff
@@ -4110,6 +4110,51 @@ var IndexQueries = []ScriptTest{
 			},
 		},
 	},
+	{
+		// MySQL allows creating multiple indexes over the same set of columns. This isn't generally
+		// useful, but some customers need this support. For example, generated migration code from
+		// Django can create cases that require this: https://github.com/dolthub/dolt/issues/8254
+		Name: "multiple indexes over same set of columns",
+		SetUpScript: []string{
+			"CREATE TABLE `t0` (`id` char(32) NOT NULL PRIMARY KEY, `col1` varchar(255) NOT NULL, `col2` varchar(255) NOT NULL);",
+		},
+		Assertions: []ScriptTestAssertion{
+			// Add two indexes over the same column set to t0
+			{
+				Query:    "ALTER TABLE t0 ADD CONSTRAINT unique_1 UNIQUE(col1, col2);",
```

Comment:
Test the auto generated unique key names; I don't think they work.

```
alter table t add unique index (i);
alter table t add unique index (i);
alter table t add unique index (i);
show create table t;
```

Reply:
ooh, that's a great suggestion, thanks!

Reply:
Added in c3ff19373fcb09f0e0d007eb655c14111205e9a8